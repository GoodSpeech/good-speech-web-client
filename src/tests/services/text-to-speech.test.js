import {
  pause,
  resume,
  cancel,
  speak
} from '../../services/text-to-speech';

describe('Text to speech', () => {
  describe('pause', () => {
    it('should call pause method', () => {
      window.speechSynthesis = {
        pause: jest.fn()
      };
      pause();
      expect(window.speechSynthesis.pause.mock.calls.length).toBe(1);
    });
  });

  describe('resume', () => {
    it('should call resume method', () => {
      window.speechSynthesis = {
        resume: jest.fn()
      };
      resume();
      expect(window.speechSynthesis.resume.mock.calls.length).toBe(1);
    });
  });

  describe('cancel', () => {
    it('should call cancel method', () => {
      window.speechSynthesis = {
        cancel: jest.fn()
      };
      cancel();
      expect(window.speechSynthesis.cancel.mock.calls.length).toBe(1);
    });
  });

  describe('speak', () => {
    it('should call speak method', () => {
      window.speechSynthesis = {
        cancel: jest.fn(),
        speak: jest.fn(),
      };
      window.SpeechSynthesisUtterance = function() {};
      speak('hello world', 'es-AR');
      expect(window.speechSynthesis.cancel.mock.calls.length).toBe(1);
      expect(window.speechSynthesis.speak.mock.calls).toMatchSnapshot();
    });
  });

});