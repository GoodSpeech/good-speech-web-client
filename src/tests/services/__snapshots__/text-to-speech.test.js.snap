// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text to speech speak should call speak method 1`] = `
Array [
  Array [
     {
      "lang": "es-AR",
      "onend": [Function],
      "onerror": [Function],
      "onstart": [Function],
      "pitch": 1,
      "rate": 1,
      "text": "hello world",
      "voiceURI": "native",
      "volume": 1,
    },
  ],
]
`;
